src/run.rs

```diff
@@ -95,12 +111,40 @@ pub fn image_ids() -> io::Result<HashSet<String>> {
             output
                 .lines()
                 .map(|line| line.trim().to_owned())
-                .filter(|line| !line.is_empty())
+                .filter_map(|line| {
+                    if line.is_empty() {
+                        let tab_index = line.find('\t').unwrap();
```

Comment:
If for some reason Docker outputs some garbage, it would be strongly preferred to return an `io::Error` over `panic`-ing (which is what `unwrap` would do). Fortunately, the return type already supports this with `io::Result`.

Since Docuum is meant to be a long running process that essentially never crashes (except under extreme circumstances like the system running out of memory), we want to be extremely judicious with any code that might panic. In particular, we generally only `unwrap` in situations where the internal invariants guarantee it to be safe (with a comment providing justification). We assume that everything external to the program (e.g., `docker` invocations) can and will fail for whatever reason (e.g., maybe the Docker daemon isn't running).

It's a little trickier to write the code this way, but it's worth it.